static/scripts/rewards/web3/connect-wallet.ts

```diff
@@ -18,18 +22,148 @@ export async function connectWallet(): Promise<JsonRpcSigner | null> {
+  const altSymbol2 = app.networkId === 1 ? "Eth" : "xDai";
+
+  if (mediaQuery.matches) {
+    /**
+     * https://github.com/MetaMask/metamask-mobile/issues/9519
+     *
+     * Until this is resolved it is not possible for us to add a network on mobile
+     * so we'll show a toast suggesting they do it manually
+     */
+
+    toaster.create("info", `${url}`, Infinity);
+    return;
+  }
+
+  // It will not work unless the symbols match, so we try them all
+  for (const _symbol of [symbol, altSymbol, altSymbol2]) {
+    // this does not work on mobile yet
+    await addProvider(provider, url, _symbol, app.networkId);
+  }
+}
+
+async function addProvider(provider: ethers.providers.Web3Provider, url: string, symbol: string, chainId: number | null) {
+  const _chainId = chainId || (await provider.getNetwork()).chainId;
+  try {
+    await provider.send("wallet_addEthereumChain", [
+      {
+        chainId: `0x${_chainId.toString(16)}`,
+        chainName: _chainId === 1 ? "Ethereum" : "Gnosis",
+        nativeCurrency: {
+          name: _chainId === 1 ? "ETH" : "XDAI",
+          symbol,
+          decimals: 18,
+        },
+        rpcUrls: [url],
+        blockExplorerUrls: [`https://${_chainId === 1 ? "etherscan" : "gnosisscan"}.io`],
+      },
+    ]);
+  } catch {
+    console.error("Failed to add network");
+  }
+}
+
+async function stressTestWalletRpc(provider: ethers.providers.Web3Provider) {
+  const success: Promise<string | boolean>[] = [];
+
+  for (let i = 0; i < 10; i++) {
+    success.push(testNonceBitmapEthCall(provider));
+  }
+
+  // if the test takes too long, we'll just assume it's not working
+  const timeoutPromise = new Promise<[false]>((resolve) => {
+    setTimeout(() => {
+      resolve([false]);
+    }, 7000);
+  });
+
+  const results = await Promise.race([Promise.all(success), timeoutPromise]);
+
+  return results.filter((s) => s === "0x" + "00".repeat(32)).length > 9 && results.filter((s) => s === false).length < 1;
+}
+
+async function testNonceBitmapEthCall(provider: ethers.providers.Web3Provider) {
+  try {
+    return await provider.send("eth_call", [
+      {
+        to: "0x000000000022D473030F116dDEE9F6B43aC78BA3",
+        // input works for desktop, needs to be data for mobile
+        data: "0x4fe02b44000000000000000000000000d9530f3fbbea11bed01dc09e79318f2f20223716001fd097bcb5a1759ce02c0a671386a0bbbfa8216559e5855698a9d4de4cddea",
+      },
+      "latest",
+    ]);
+  } catch {
+    // if the call fails, we'll assume the RPC is not working
+  }
+}
+
 function connectErrorHandler(error: unknown) {
   if (error instanceof Error) {
     console.error(error);
     if (error?.message?.includes("missing provider")) {
       // mobile browsers don't really support window.ethereum
-      const mediaQuery = window.matchMedia("(max-width: 768px)");
 
       if (mediaQuery.matches) {
         toaster.create("warning", "Please use a mobile-friendly Web3 browser such as MetaMask to collect this reward", Infinity);
```

Comment:
I'm surprised we used media queries in the typescript code at all. It really doesn't seem to be the best way to detect if it's mobile browser.

Reply:
What would you suggest instead of? afaik it's the most optimal way to detect

Reply:
I remember using this approach https://stackoverflow.com/a/11381730/23135269